frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,45 @@
+import { IconColor } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+
+import { Button } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: React.ReactNode;
+  isDisplayed: boolean;
+  title: string;
+  onClose: () => void;
+};
+
+const Modal: React.FC<Properties> = ({
```

Comment:
Let's add useHandleClickOutside hook here

Reply:
Let me tag @Dmytro-Kucherenko here, since its his component initially, want to make sure he also knows about the change

Reply:
Okay, I got it